src/hooks/useDomainSearch.tsx

```diff
@@ -0,0 +1,54 @@
+import type { Domain } from '@zero-tech/zns-sdk';
+
+import { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useZnsSdk } from './useZnsSdk';
+import { getDomainId } from '../helpers/domains';
+import { MIN_DOMAIN_NAME_LENGTH } from '../constants/domains';
+
+export interface UseDomainSearchReturn {
+	isLoading: boolean;
+	pattern: string;
+	setPattern: (pattern: string) => void;
+	exactMatch?: Domain;
+}
+
+export const useDomainSearch = (): UseDomainSearchReturn => {
+	// SDK
+	const sdk = useZnsSdk();
+
+	// State
+	const [pattern, setPattern] = useState<string>('');
+
+	// Query
+	const { isLoading, data: exactMatch } = useQuery(
+		`buy-domains-search-${pattern}`,
+		async () => {
+			if (pattern.length >= MIN_DOMAIN_NAME_LENGTH) {
```

Comment:
It may be worth throwing an error if the domain name is less than minimum length - I think this will make it more explicit that the query was invalid, rather than returning `undefined`.

Reply:
Outdated method! used new SDK functions!!!!!!